core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -205,6 +207,13 @@ private[spark] class HeartbeatReceiver(sc: SparkContext, clock: Clock)
             // Note: we want to get an executor back after expiring this one,
             // so do not simply call `sc.killExecutor` here (SPARK-8119)
             sc.killAndReplaceExecutor(executorId)
+            // In case of the executors which are not gracefully shut down, we should remove
+            // lost executors from CoarseGrainedSchedulerBackend manually here (SPARK-27348)
+            sc.schedulerBackend match {
+              case backend: CoarseGrainedSchedulerBackend =>
```

Comment:
how about other backed?

Reply:
For all 3 kinds of SchedulerBackend, both `CoarseGrainedSchedulerBackend` and `StandaloneSchedulerBackend` can be handled here, the last one, `LocalSchedulerBackend`, is unnecessary because this PR wants to fix the problem of clearing lost executor depends on `onDisconnect` in distributed scheduler backend, while `LocalSchedulerBackend` has no such problem. Please correct me if I'm wrong.